### ISSUE TITLE
[Dev support]: Help with configuring authentication

### ISSUE THREAD

**Issue by yikwen** [microsoft/teams-ai#1927]
Hi I've implemented an Oauth 2 Generic Provider on my bot service resource in Azure and tested the connection successfully. After configuring authentication as per documentation [here](https://github.com/microsoft/teams-ai/blob/main/getting-started/CONCEPTS/USER-AUTH.md) with `auto` set to True, clicking on the Sign In button the adaptive card results in error "Something went wrong. Please try again."

![image](https://github.com/user-attachments/assets/5c3d6eeb-fe3d-478e-a1ca-540c9e85303f)

On mobile, this is what I see after pressing the Sign In button:

![image](https://github.com/user-attachments/assets/a7ecf673-5d0f-41ec-bf9b-c95d89935daa)

**Comment by yikwen** [comment]
Testing the bot via web chat on the Azure Portal, the sign in button works - it opens the browser to the application's sign in page, OAuth consent screen, and once done a code is generated. Pasting the code back in the chat doesn't do anything.

![image](https://github.com/user-attachments/assets/7c571789-74c6-4592-89c8-d7c003b77770)

**Comment by yikwen** [comment]
@corinagum yes using Python. You're referring to #1934? Are they the same issue with the same causes, given that the other one is SSO with Entra ID, and this one is a generic OAuth provider?

**Comment by SubbaReddi** [comment]
@yikwen : Can you specify the detailed reproduce steps to create above scenario?

**Comment by yikwen** [comment]
@SubbaReddi 

1. Register an app with the service provider (in my case, https://talenox.github.io/api-doc/#introduction)
2. Configure bot service with an Oauth2 Generic Provider connection following documentation [here](https://learn.microsoft.com/en-us/azure/bot-service/bot-builder-concept-identity-providers?view=azure-bot-service-4.0&tabs=adv2%2Ca2gp#oauth-generic-providers)
![image](https://github.com/user-attachments/assets/d2b1e292-cc70-4d78-8a0e-e82007e44f27)
3. Added auth options to my app
![image](https://github.com/user-attachments/assets/c93741cd-b416-4195-88a9-36d3bcfb8f62)
4. And test code to trigger sign in
![image](https://github.com/user-attachments/assets/ebc40784-56b3-4e6d-b1e9-6cd6f851c5ed)

That's all.

**Comment by SubbaReddi** [comment]
@yikwen : Issue is not reproducible for me with the above steps. Something went wrong error message is not displaying.
Popup window is opening to provide consent and then window closing. But finally leading to the same issue as reported [here ](https://github.com/microsoft/teams-ai/issues/1934)

**Comment by yikwen** [comment]
@SubbaReddi just curious, what service provider did you register your application with, and what version of the Python SDK were you working with? I was on 1.3.0, and if I could try the same service provider as the one you tested with then we can see if the behaviour's the same.

**Comment by SubbaReddi** [comment]
@yikwen : I used talenox service provided as you mentioned above. I was using Teams sdk 1.2.2 with this sample: https://github.com/microsoft/teams-ai/tree/main/python/samples/06.auth.oauth.bot

**Comment by yikwen** [comment]
> @yikwen : I used talenox service provided as you mentioned above. I was using Teams sdk 1.2.2 with this sample: https://github.com/microsoft/teams-ai/tree/main/python/samples/06.auth.oauth.bot

@SubbaReddi i was on v1.3.0

**Comment by SubbaReddi** [comment]
@yikwen : Is it possible to try with above sample with Teams sdk 1.2.2? Currently i have a different blocker to validate with v1.3.0

**Comment by yikwen** [comment]
@SubbaReddi unfortunately no, I've configured my code with v1.4.0 features and changing to v1.2.2 would require quite a bit of things to change in my code before I can test OAuth.

**Comment by yikwen** [comment]
@SubbaReddi FYI still not working even with latest Teams client and teams-ai 1.5.0, same behaviour as before as well as #1934

**Comment by corinagum** [comment]
@yikwen just jumping in to say, we just released 1.7.0 (latest was 1.6.1 before today). Please try updating the SDKto that.

**Comment by corinagum** [comment]
I was unable to reproduce the token exchange problem. @yikwen could you spin up a repo that you can share that reproduces the issue?

**Comment by yikwen** [comment]
@corinagum I just simply deployed the [oauth bot sample](https://github.com/microsoft/teams-ai/tree/main/python/samples/06.auth.oauth.bot) and configuring it to make use of the MS Graph connection that I've setup (and tested successfully) on my Azure Bot. Did not modify nor create any custom code.
